File: lists/stack.py
class Node:
	def __init__(self, value):
		self.value = value
		self.next = None

class Stack:
	def __init__(self):
		self.head = None
		self.stack_length = 1

	def push(self, item):
		self.head.next = item
		self.stack_length += 1

	def pop(self):
		self.items.pop()

	def printStack(self):
		print(self.items)

	def size(self):
		print(self.stack_length)

s = Stack()

s.head = Node(1)

s.push(Node(2))

print(s.head.value)
print(s.head.next.value)
s.size()

File: lists/stack.ts

interface NodeProperties {
	value: number;
	next?: NodeProperties;
}

interface StackProperties {
	head?: NodeProperties;
	length: number;
	pop: () => void;
	push: () => void;
	printStack: () => void;
	size: () => void;
}

class Stack<StackProperties> {

	private length = 1;

	constructor(public head?: NodeProperties) {}

	pop = () => {
		this.head = this.head.next;
	}

	push = (node: NodeProperties) => {
		const auxNode = this.head;
		this.head = node;
		this.head.next = auxNode;
		this.length += 1;
	}

	printStack = () => {
		let auxNode = this.head;
		while(auxNode) {
			console.log(auxNode.value);
			auxNode = auxNode.next;
		}
	}

	size = () => {
		console.log(`Size of stack is: ${this.length}`);
	}
}

class MyNode<NodeProperties> {

	constructor(public value: number) {}

}

const head = new MyNode(1)

const stack = new Stack(head);

const newNode = new MyNode(2);

stack.push(newNode);

stack.size();

// stack.printStack();

stack.push(new MyNode(3));

stack.pop();

stack.push(new MyNode(4));

stack.printStack();
